content/post/2019-12-26-r-hub-usage.Rmarkdown

```diff
@@ -0,0 +1,133 @@
+---
+title: R-hub usage in a few figures
+date: '2019-12-26'
+slug: r-hub-usage
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE,
+                      cache = TRUE,
+                      warning = FALSE)
+```
+
+
+Earlier this year when launching this blog we [explained why R package developers should care about R-hub](/2019/03/26/why-care/). So, does anyone care? :wink: Yes! Let's summarize R-hub usage.
+
+## Usage data
+
+As specified in our [use terms](https://builder.r-hub.io/terms.html) we do not store user data. However we do store some usage data. In the table loaded below, emails and packages are unidentified -- but you can know whether an email or package comes up several times. 
+
+```{r, echo = FALSE}
+my_not_portable_path <- "/home/maelle/Documents/R-hub/rhub-usage/rhub-builds.rds"
+```
+
+```{r}
+builds <- readRDS(my_not_portable_path)
+builds <- dplyr::mutate_at(builds, c("submitted", "started"), anytime::anytime)
+str(builds)
+```
+
+## A recent increase in usage
+
+### Towards 1,000 builds a week? 
+
+```{r usage-week, fig.cap="Weekly count of builds on R-hub package builder, showing an slow increase until mid 2018 then a steeper increase to a little less than 1,000 builds a week"}
+library("ggplot2")
+library("magrittr")
+dplyr::mutate(builds, 
+             week = as.Date(cut(submitted, "week"))) %>%
+  dplyr::count(week) %>%
+ggplot(aes(week, n)) +
+  geom_point() +
+  geom_smooth() +
+  ylab("No. of buids") +
+  xlab("Time (weeks)") +
+  hrbrthemes::theme_ipsum(base_size = 16,
+                          axis_title_size = 16)
+```
+
+When plotting the weekly count of builds as below, it is quite clear that usage stepped up at the end of last year. A delayed effect of the [RStudio webinar about R-hub](https://resources.rstudio.com/the-essentials-of-data-science/r-hub-overview-ga-bor-csa-rdi)?
+
+### Number of packages built per week
+
+```{r usage-week-pkg, fig.cap="Weekly count of builds on R-hub package builder, showing an increase, then a stagnation in 2018, then a steeper increase since the end of 2018, to about 125 packages a week. WHAT TO SAY ABOUT THE HIGH NUMBER OF PACKAGES SOME WEEKS IN 2017?"}
```

Comment:
For a while RStudio was using R-hub to build binaries for some of their hosted services, maybe shinyapps.io, maybe something else. That could be it.